File: 01/flavien/AoC/index-1.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final input = await File('./input.txt').readAsString();
  final reduced = LineSplitter().convert(input).map((line) {
    final lineDigits =
        RegExp(r'\d').allMatches(line).map((match) => match.group(0));

    return int.parse('${lineDigits.first}${lineDigits.last}');
  }).fold(0, (accumulator, currentValue) => accumulator + currentValue);

  print(reduced);
}

File: 01/flavien/AoC/index-1.mjs
import { readFile } from "fs/promises"
console.log(
  (await readFile("./input.txt", { encoding: "utf8" }))
    .split("\n")
    .map((line) => {
      const lineDigits = Array.from(
        line.matchAll(new RegExp(/\d/g)),
        (m) => m[0]
      )

      return Number(`${lineDigits.at(0)}${lineDigits.at(-1)}`)
    })
    .reduce((accumulator, currentValue) => (accumulator += currentValue), 0)
)
